src/Backend/Modules/MediaLibrary/Js/MediaLibraryHelper.js

```diff
@@ -188,6 +189,30 @@ jsBackend.mediaLibraryHelper.group = {
       // get media for this folder
       jsBackend.mediaLibraryHelper.group.getMedia()
     })
+
+    $('#searchMedia').on('click', function(e){
+      e.preventDefault();
+      e.stopPropagation();
+
+      mediaFolderId = $('#mediaFolders').val();
+      searchQuery = $('[name=query]').val();
+
+      jsBackend.mediaLibraryHelper.group.clearMediaCache(mediaFolderId);
+      jsBackend.mediaLibraryHelper.group.getMedia()
+    })
+
+    $('[name=query]').on('keyup', function(e){
+      if (e.keyCode == 13) {
```

Comment:
Personal preference: because we already use ajax here, the submit button feels a bit weird to me. Maybe its better to just have a search input field without submit button, and let it immediately filter when you start typing? As that feels more natural to me in datagrids. 

However, you would need to debounce the input being typed in, but there's a helper debounce function in the `utils.js`, like this example uses when typing in the page title and it's auto-generating the meta URL preview while typing:

```javascript
// jquery.backend.js:50
$element.bind('keyup input', utils.events.debounce(calculateMeta, 400))
```

Reply:
> Not sure if listening to the enter key (and submit button) is the best solution here 👀

The modal is loaded inside the existing form. When someone presses the enter key the form is posted and not the search form. I noticed it myself, therefor I added the check.



> Personal preference: because we already use ajax here, the submit button feels a bit weird to me. Maybe its better to just have a search input field without submit button, and let it immediately filter when you start typing? As that feels more natural to me in datagrids.
> 
> However, you would need to debounce the input being typed in, but there's a helper debounce function in the `utils.js`, like this example uses when typing in the page title and it's auto-generating the meta URL preview while typing:
> 
> ```js
> // jquery.backend.js:50
> $element.bind('keyup input', utils.events.debounce(calculateMeta, 400))
> ```

This could be an option indeed, I'll check this.

Reply:
After some more testing I noticed that the enter button check isn't required. I also implemented the debounce function, didn't know about its existence but works nice.

I realy would like to keep de search button for user experience.

Reply:
Cool :D Thank you!